### DISCUSSION TITLE
function call如何避免回答到一半就卡住的问题？

### DISCUSSION THREAD

**Discussion by si1mon2** [zai-org/ChatGLM3#730] [General]
![1](https://github.com/THUDM/ChatGLM3/assets/8703097/091817ae-37b2-44f3-a2fa-ee278a1af1ff)

如图，在工具调用的时候，回答总是会在出现冒号（举例说明）的地方就卡住，需要继续问下去才回答全，想请教该如何解决这个问题？

**Comment by stivensss** [comment]
我也遇到这个问题

**Reply by zRzRzRzRzRzRzR** [reply]
使用命令行调用呢， 这个界面的工具调用也不对啊

**Reply by si1mon2** [reply]
初衷是希望能让非专业人士也能通过上述的多轮问询调用他们想选的function，非专业人士未必熟悉命令行的使用，所以有上图这样交互方式，多轮问询以后再把要选的function和参数传递进去（图上展示的是有具体步骤的解释以后，后面再调用对应的函数）。经测试以后发现如果定义的function个数太多，就会出现上面这样卡住的情况，如果注释掉一部分不常用的function，就能正常回答，这是否和每次接收/回复字段的长度limit有关系？是否有更通用的解决方法？

**Reply by zRzRzRzRzRzRzR** [reply]
基础代码只有提供一个能用的UI界面，这个UI界面逻辑很简陋，根据我看到截图情况来看，模型甚至没有输出符合格式的funtioncall 内容，关于多个function无法识别，开源的6b模型确实有这个情况，因为没法处理那么多工具，商业版本具有这个能力。可以使用我们的商业版本。
如果你想使用开源版本， 那确实只能降低被模型调用工具的数量（大概最多到3个），你可以通过其他工程算法实现来初步筛选掉不可能使用的工具在给模型，这些功能在开源仓库中没有提供代码，需要自己完成或寻求商业解决方案

**Reply by si1mon2** [reply]
明白了，谢谢解答
